source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -195,6 +196,18 @@ ClusterFactory::createClusterWithConfig(
   auto new_cluster = std::make_shared<Cluster>(
       cluster_config, proto_config, context.runtime(), cache_manager_factory, context.localInfo(),
       socket_factory_context, std::move(stats_scope), context.addedViaApi());
+
+  auto& options = new_cluster->info()->upstreamHttpProtocolOptions();
+
+  if (!proto_config.allow_insecure_cluster_options()) {
+    if (!options.has_value() ||
+        (!options.value().auto_sni() || !options.value().auto_san_validation())) {
+      throw EnvoyException(
+          "dynamic_forward_proxy cluster must have auto_sni and auto_san_validation true unless "
+          "allow_insecure_cluster_options is set.");
+    }
```

Comment:
I think you can remove this check from above (and just set the legacy option) since this should check both cases after the cluster is created?